cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs

```diff
@@ -1290,14 +1512,123 @@ instance Show peerConn => ToObject (DebugPeerSelection Socket.SockAddr peerConn)
              ]
 
 instance ToObject (PeerSelectionActionsTrace Socket.SockAddr) where
-  toObject _verb ev =
-    mkObject [ "kind" .= String "PeerSelectionAction"
-             , "event" .= show ev ]
+  toObject _verb (PeerStatusChanged ps) =
+    mkObject [ "kind" .= String "PeerStatusChanged"
+             , "peerStatusChangeType" .= show ps
+             ]
+  toObject _verb (PeerStatusChangeFailure ps f) =
+    mkObject [ "kind" .= String "PeerStatusChangeFailure"
+             , "peerStatusChangeType" .= show ps
+             , "error" .= show f
+             ]
+  toObject _verb (PeerMonitoringError cs s) =
+    mkObject [ "kind" .= String "PeerMonitoringError"
+             , "connectionId" .= toObject _verb cs
+             , "error" .= show s
+             ]
+  toObject _verb (PeerMonitoringResult cs wf) =
+    mkObject [ "kind" .= String "PeerMonitoringResult"
+             , "connectionId" .= toObject _verb cs
+             , "withProtocolTemp" .= show wf
+             ]
 
 instance ToObject PeerSelectionCounters where
   toObject _verb ev =
     mkObject [ "kind" .= String "PeerSelectionCounters"
-             , "event" .= show ev ]
+             , "coldPeers" .= coldPeers ev
+             , "warmPeers" .= warmPeers ev
+             , "hotPeers" .= hotPeers ev
+             ]
+
+instance ToJSON ProtocolLimitFailure where
+  toJSON (ExceededSizeLimit (ClientAgency _)) =
+    Aeson.object [ "kind" .= String "ProtocolLimitFailure"
+                 , "exceededSizeLimit" .= String "ClientAgency"
```

Comment:
It would be useful to actually show the state token.  We provide `Show` instances for `PeerHasAgency`,  both `ExceededSizeLimit` and `ExceededTimeLimit` provide them when you pattern match on them.  You can have one pattern match for `ExceededSizeLimit tok` and then use `show tok`.